src/tests/roc_library/test_sender_receiver.cpp

```diff
@@ -11,534 +11,282 @@
-                                    recv_repair_config_.bind_address.port())
-              == 0);
-    }
+    test::Receiver receiver(context, receiver_conf, sample_step, test::FrameSamples,
+                            Flags);
 
-    ~Proxy() {
-        CHECK(roc_endpoint_deallocate(input_source_endp_) == 0);
-        CHECK(roc_endpoint_deallocate(input_repair_endp_) == 0);
-    }
+    test::Proxy proxy(receiver.source_endpoint(), receiver.repair_endpoint(),
+                      test::SourcePackets, test::RepairPackets);
 
-    const roc_endpoint* source_endpoint() const {
-        return input_source_endp_;
-    }
+    test::Sender sender(context, sender_conf, proxy.source_endpoint(),
+                        proxy.repair_endpoint(), sample_step, test::FrameSamples, Flags);
 
-    const roc_endpoint* repair_endpoint() const {
-        return input_repair_endp_;
-    }
+    sender.start();
+    receiver.run();
+    sender.stop();
+    sender.join();
+}
 
-private:
-    virtual void write(const packet::PacketPtr& pp) {
-        pp->udp()->src_addr = send_config_.bind_address;
+TEST(sender_receiver, separate_context) {
+    enum { Flags = 0 };
 
-        if (pp->udp()->dst_addr == recv_source_config_.bind_address) {
-            pp->udp()->dst_addr = receiver_source_endp_;
-            source_queue_.write(pp);
-        } else {
-            pp->udp()->dst_addr = receiver_repair_endp_;
-            repair_queue_.write(pp);
-        }
+    init_config(Flags);
 
-        for (;;) {
-            const size_t block_pos = pos_ % (n_source_packets_ + n_repair_packets_);
-
-            if (block_pos < n_source_packets_) {
-                if (!send_packet_(source_queue_, block_pos == 1)) {
-                    return;
-                }
-            } else {
-                if (!send_packet_(repair_queue_, false)) {
-                    return;
-                }
-            }
-        }
-    }
+    test::Context recv_context, send_context;
 
-    bool send_packet_(packet::IReader& reader, bool drop) {
-        packet::PacketPtr pp = reader.read();
-        if (!pp) {
-            return false;
-        }
-        pos_++;
-        if (!drop) {
-            writer_->write(pp);
-        }
-        return true;
-    }
+    test::Receiver receiver(recv_context, receiver_conf, sample_step, test::FrameSamples,
+                            Flags);
 
-    netio::UdpSenderConfig send_config_;
+    test::Sender sender(send_context, sender_conf, receiver.source_endpoint(),
+                        receiver.repair_endpoint(), sample_step, test::FrameSamples,
+                        Flags);
 
-    roc_endpoint* input_source_endp_;
-    roc_endpoint* input_repair_endp_;
+    sender.start();
+    receiver.run();
+    sender.stop();
+    sender.join();
+}
 
-    netio::UdpReceiverConfig recv_source_config_;
-    netio::UdpReceiverConfig recv_repair_config_;
+TEST(sender_receiver, multiple_senders_one_receiver) {
```

Comment:
This test sequentially runs one sender and then another. It's a good idea to add such a test, but we also need a test that runs two senders in parallel. It would be a more complicated task because we'll have to adjust receiver logic accordingly.

I suggest to rename this test to `multiple_senders_one_receiver_sequential` and then add a new test `multiple_senders_one_receiver_simultaneous`. I think it's better to add the new test in a new PR because this one is too big already. In this PR we can just rename the test.